features/include/pcl/features/impl/scurv.hpp

```diff
@@ -0,0 +1,864 @@
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT, typename PointOutT> void
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::positioningAtCenter (pcl::PointCloud<pcl::PointNormal> &cloudToBeNormalized)
+{
+  for (size_t k = 0;  k < cloudToBeNormalized.points.size(); k++)
+  {
+    cloudToBeNormalized.points[k].x -= theMeanStruct.theMeanX;
+    cloudToBeNormalized.points[k].y -= theMeanStruct.theMeanY;
+    cloudToBeNormalized.points[k].z -= theMeanStruct.theMeanZ;
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT, typename PointOutT> Eigen::MatrixXd
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::svd_orthoBasisForTheNullSpace (const Eigen::MatrixXd &theMatrix)
+{
+  //svd and orth. basis for the null space
+        
+  Eigen::JacobiSVD<Eigen::MatrixXd> svd(theMatrix, Eigen::ComputeFullV);
+  Eigen::MatrixXd S(1, 3);
+  S = Eigen::MatrixXd::Zero(1, 3);
+  S(0, 0) = svd.singularValues()(0,0);
+  Eigen::MatrixXd toReturn(3, 2);
+  for (size_t k =0; k < 3; k++)
+  {
+    toReturn(k, 0) = svd.matrixV() (k, 1);
+    toReturn(k, 1) = svd.matrixV() (k, 2);
+  }
+  
+  return toReturn;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT, typename PointOutT> double
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::pchst (double arg1, double arg2)
+{
+  double value;
+
+  if ( arg1 == 0.0 )
+  {
+    value = 0.0;
+  }
+  else if ( arg1 < 0.0 )
+  {
+    if ( arg2 < 0.0 )
+    {
+      value = 1.0;
+    }
+    else if ( arg2 == 0.0 )
+    {
+      value = 0.0;
+    }
+    else if ( 0.0 < arg2 )
+    {
+      value = -1.0;
+    }
+  }
+  else if ( 0.0 < arg1 )
+  {
+    if ( arg2 < 0.0 )
+    {
+      value = -1.0;
+    }
+    else if ( arg2 == 0.0 )
+    {
+      value = 0.0;
+    }
+    else if ( 0.0 < arg2 )
+    {
+      value = 1.0;
+    }
+  }
+
+  return value;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT, typename PointOutT> void
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::spline_pchip_set (int n, std::vector<double> x, std::vector<double> f, std::vector<double> &d)
+{
+  double del1, del2, dmax, dmin, drat1, drat2, dsave, h1, h2, hsum, hsumt3, temp, w1, w2;
+  int i, ierr, nless1;
+//
+//  Check the arguments.
+//
+  if ( n < 2 )
+  {
+    ierr = -1;
+    std::cerr << "\n";
+    std::cerr << "SPLINE_PCHIP_SET - Fatal error!\n";
+    std::cerr << "  Number of data points less than 2.\n";
+    exit ( ierr );
```

Comment:
This is not ok. You cannot exit the application for the user. Throw an exception and let the user decide what to do with it.